src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java

```diff
@@ -0,0 +1,156 @@
+import org.breedinginsight.services.ProgramService;
+import org.breedinginsight.services.exceptions.DoesNotExistException;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.UUID;
+
+@Slf4j
+@Controller
+@Secured(SecurityRule.IS_AUTHENTICATED)
+public class BrAPIV2Controller {
+
+    private final SecurityService securityService;
+    private final ProgramService programService;
+
+    @Inject
+    public BrAPIV2Controller(SecurityService securityService, ProgramService programService) {
+        this.securityService = securityService;
+        this.programService = programService;
+    }
+
+
+    @Get(BrapiVersion.BRAPI_V2 + "/serverinfo")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Secured(SecurityRule.IS_ANONYMOUS)
+    public BrAPIServerInfoResponse serverinfo() {
+        BrAPIServerInfo serverInfo = new BrAPIServerInfo();
+        serverInfo.setOrganizationName("Breeding Insight Platform");
+        serverInfo.setServerName("bi-api");
+        serverInfo.setContactEmail("bidevteam@cornell.edu");
+        serverInfo.setOrganizationURL("breedinginsight.org");
+
+        return new BrAPIServerInfoResponse().result(serverInfo);
+    }
+
+    //TODO override program serverinfo endpoint to replace some of the data points (contactEmail, organizationName, organizationURL, etc)
+
+    @Get("/${micronaut.bi.api.version}/programs/{programId}" + BrapiVersion.BRAPI_V2 + "/{+path}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roleGroups = {ProgramSecuredRoleGroup.ALL})
+    public HttpResponse<?> getCatchall(@PathVariable("path") String path, @PathVariable("programId") UUID programId, HttpRequest<String> request) {
+        return executeRequest(path, programId, request, "GET");
+    }
+
+    @Post("/${micronaut.bi.api.version}/programs/{programId}" + BrapiVersion.BRAPI_V2 + "/{+path}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roleGroups = {ProgramSecuredRoleGroup.ALL})
+    public HttpResponse<String> postCatchall(@PathVariable("path") String path, @PathVariable("programId") UUID programId, HttpRequest<String> request) {
+        return executeRequest(path, programId, request, "POST");
+    }
+
+    @Put("/${micronaut.bi.api.version}/programs/{programId}" + BrapiVersion.BRAPI_V2 + "/{+path}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roleGroups = {ProgramSecuredRoleGroup.ALL})
+    public HttpResponse<String> putCatchall(@PathVariable("path") String path, @PathVariable("programId") UUID programId, HttpRequest<String> request) {
+        return executeRequest(path, programId, request, "PUT");
+    }
+
+    private HttpResponse<String> executeRequest(String path, UUID programId, HttpRequest<String> request, String method) {
+        AuthenticatedUser actingUser = securityService.getUser();
+        System.out.println(actingUser.getId());
```

Comment:
Can probably get rid of these printlns, maybe use log if it's important?

Reply:
Changed to logging statements, good catch!